writeengine/wrapper/writeengine.cpp

```diff
@@ -1681,6 +1681,13 @@ int WriteEngineWrapper::insertColumnRecs(const TxnID& txnid,
         // Write row(s) to database file(s)
         //----------------------------------------------------------------------
         rc = writeColumnRec(txnid, cscColTypeList, colStructList, colOldValueList, rowIdArray, newColStructList, colNewValueList, tableOid, useTmpSuffix); // @bug 5572 HDFS tmp file
+
+        if (rc == NO_ERROR)
+        {
+            rc = cacheutils::flushPrimProcCache();
```

Comment:
I wonder if you understand how heavy is the operation for a big caches?

Reply:
I understand your concern. I have updated to only flush the dictionary oids from the cache